File: milk/src/test/java/com/yq/milk/Super.java
package com.yq.milk;

/**
 * Created by king on 2018/2/23.
 */

public abstract class Super {

    int age = 19;
    String name;

    public Super(){
        this.name = way();
    }

    public void say(String s){
        System.out.println( s + " "+age+" "+ name);
    }

    public abstract String way();
}

File: milk_kotlin/src/test/java/com/yq/milk_kotlin/Super.kt
package com.yq.milk_kotlin

/**
 * Created by king on 2018/2/23.
 */
abstract class Super {
    val age = 19
    abstract var name : String

    open fun say(s: String): Super{
        print("$s $name $age")
        return this
    }
}